Functional verification using SVA and UVM

atefitani opened the pull request:
Hello @mjlyons, it was fun to verify your design! (although it is hardware-proved 😄)

The following was achieved:

Two verification methods were applied: Assertion-based and UVM.
15 Assertions and Coverages were designed to test all the DUT signals.
200k bytes were randomly stimulated.
In UVM, we implemented [Sunburst scoreboard.](https://sunburst-design.com/papers/CummingsSNUG2013SV_UVM_Scoreboards.pdf)
Both methods achieved a 100% pass rate and more than 96% coverage.
A detailed report with diagrams and a conclusion was written and attached.
please note: this was a verification course project.

atefitani committed:
Adding SVA and UVM

atefitani committed:
Adding the report and readme file

atefitani committed:
Update ReadMe.txt

atefitani committed:
Create README.md

atefitani committed:
deleting readme.txt

atefitani committed:
delete ReadMe.txt from master

atefitani committed:
Merge remote-tracking branch 'origin/verification'

atefitani committed:
Update README.md

mjlyons commented:
@atefitani: Thanks for submitting this! It's easily been a decade since I've touched this code (or any hardware development, for that matter), so I don't have a great way to review this. However, if you'd like to add a disclaimer like the following to your `verification/README.md` I'd be happy to merge it:

```
DISCLAIMER: The code in `verification/` was authored by @atefitani. This code was not authored or verified by the original author (@mjlyons).
```

Thanks for your work and contribution! Hopefully it helps others :)

Also, I assume #2 was closed and this is a resubmit of that code?

atefitani committed:
Update README.md

atefitani commented:
Thank you @mjlyons  for your comment, it was fun verifying the design :)
I added the disclaimer to the READ.md in /verification.
Yes, #2 was closed and this is a resubmit. First, I pushed the code on a separate branch, then merged it to the master and deleted the branch.

Best regards
